app/components/SearchPopover.tsx

```diff
@@ -0,0 +1,164 @@
+import Empty from "~/components/Empty";
+import InputSearch from "~/components/InputSearch";
+import Placeholder from "~/components/List/Placeholder";
+import PaginatedList from "~/components/PaginatedList";
+import Popover from "~/components/Popover";
+import useStores from "~/hooks/useStores";
+import SearchListItem from "./SearchListItem";
+
+type Props = { shareId: string };
+
+function SearchPopover({ shareId }: Props) {
+  const { t } = useTranslation();
+  const { documents } = useStores();
+  const theme = useTheme();
+
+  const popover = usePopoverState({
+    placement: "bottom-start",
+    unstable_offset: [-24, 0],
+    modal: true,
+  });
+
+  const [query, setQuery] = React.useState("");
+  const searchResults = documents.searchResults(query);
+
+  const performSearch = React.useCallback(
+    async ({ query, ...options }: Record<string, any>) => {
+      if (query?.length > 0) {
+        return await documents.search(query, { shareId, ...options });
+      }
+      return undefined;
+    },
+    [documents, shareId]
+  );
+
+  const handleSearch = React.useMemo(
+    () =>
+      debounce(async (event: React.ChangeEvent<HTMLInputElement>) => {
+        const { value } = event.target;
+
+        if (value.length) {
+          popover.show();
+          setQuery(event.target.value.trim());
+        } else {
+          popover.hide();
+        }
+      }, 400),
+    [popover]
+  );
+
+  const searchInputRef = popover.unstable_referenceRef;
+  const firstSearchItem = React.useRef<HTMLAnchorElement>(null);
+
+  const handleEscapeList = React.useCallback(
+    () => searchInputRef?.current?.focus(),
+    [searchInputRef]
+  );
+
+  const handleKeyDown = (ev: React.KeyboardEvent<HTMLInputElement>) => {
+    if (ev.key === "Enter") {
+      if (searchResults?.length) {
+        popover.show();
+      }
+    }
+
+    if (ev.key === "ArrowDown" && !ev.shiftKey) {
+      if (searchResults?.length) {
+        if (ev.currentTarget.value.length === ev.currentTarget.selectionStart) {
+          popover.show();
+        }
+        firstSearchItem.current?.focus();
+      }
+    }
+
+    if (ev.key === "ArrowUp") {
+      if (popover.visible) {
+        popover.hide();
+        ev.preventDefault();
+      }
+
+      if (ev.currentTarget.value) {
+        if (ev.currentTarget.selectionEnd === 0) {
+          ev.currentTarget.selectionStart = 0;
+          ev.currentTarget.selectionEnd = ev.currentTarget.value.length;
+          ev.preventDefault();
+        }
+      }
+    }
+
+    if (ev.key === "Escape") {
+      if (popover.visible) {
+        popover.hide();
+        ev.preventDefault();
+      }
+    }
+  };
+
+  return (
+    <>
+      <PopoverDisclosure {...popover}>
+        {(props) => {
+          // props assumes the disclosure is a button, but we want a type-ahead
+          // so we take the aria props, and ref and ignore the event handlers
+          return (
+            <InputSearch
```

Comment:
![image](https://user-images.githubusercontent.com/380914/161210426-44b49c59-8baf-43b6-a185-2645b1135277.png)

Can we align vertically with the doc header?